lib/pbench/test/unit/server/test_shell_cli.py

```diff
@@ -0,0 +1,384 @@
+        ) -> str:
+            if oidc_conf:
+                return "https://oidc.example.com"
+            else:
+                raise OpenIDClient.NotConfigured()
+
+        commands = []
+
+        def run(args, cwd: Optional[str] = None) -> subprocess.CompletedProcess:
+            commands.append(args)
+            ret_val = 42 if args[0] == "gunicorn" else 0
+            return subprocess.CompletedProcess(args, ret_val)
+
+        monkeypatch.setattr(shell.site, "ENABLE_USER_SITE", user_site)
+        monkeypatch.setattr(shell, "find_the_unicorn", find_the_unicorn)
+        monkeypatch.setattr(
+            shell.OpenIDClient, "wait_for_oidc_server", wait_for_oidc_server
+        )
+        monkeypatch.setattr(subprocess, "run", run)
+
+        ret_val = shell.main()
+
+        assert ret_val == 42
+        assert not user_site or find_called[0]
+        assert len(commands) == 3, f"{commands!r}"
+        assert commands[0][0] == "pbench-create-crontab"
+        assert commands[1][0] == "crontab"
+        gunicorn_command = commands[2]
+        assert gunicorn_command[-1] == "pbench.cli.server.shell:app()", f"{commands!r}"
+        gunicorn_command = gunicorn_command[:-1]
+        if user_site:
+            assert (
+                gunicorn_command[-2:][0] == "--pythonpath"
+            ), f"commands[2] = {commands[2]!r}"
+            gunicorn_command = gunicorn_command[:-2]
+        expected_command = [
+            "gunicorn",
+            "--workers",
+            "3",
+            "--timeout",
+            "9000",
+            "--pid",
+            "/run/pbench-server/gunicorn.pid",
+            "--bind",
+            "0.0.0.0:8001",
+            "--log-syslog",
+            "--log-syslog-prefix",
+            "pbench-server",
+        ]
+        assert (
+            gunicorn_command == expected_command
+        ), f"expected_command = {expected_command!r}, commands[2] = {commands[2]!r}"
+
+    @staticmethod
+    def test_main_crontab_failed(monkeypatch, make_logger, mock_get_server_config):
+        def generate_crontab_if_necessary(*args, **kwargs) -> int:
+            return 43
+
+        monkeypatch.setattr(shell.site, "ENABLE_USER_SITE", False)
+        monkeypatch.setattr(
+            shell, "generate_crontab_if_necessary", generate_crontab_if_necessary
+        )
+
+        ret_val = shell.main()
+
+        assert ret_val == 43
+
+    @staticmethod
+    @pytest.mark.parametrize("init_db_exc", ["section", "option"])
+    def test_main_initdb_failed(
+        monkeypatch, make_logger, mock_get_server_config, init_db_exc
+    ):
+        def init_db(*args, **kwargs) -> int:
+            if init_db_exc == "section":
+                exc = NoSectionError("missingsection")
+            elif init_db_exc == "option":
+                exc = NoOptionError("section", "missingoption")
+            else:
+                raise Exception(f"Bad test parameter, {init_db_exc}")
+            raise exc
```

Comment:
In line 269, s/raise/exc =/ ?

Ditto line 375.

Reply:
Fixed.